index.mjs

```diff
@@ -1,21 +1,49 @@
 import { compileString } from "squint-cljs";
-import esbuild from "esbuild";
+import path, { dirname } from "path";
+import fs from "fs";
 
-// TODO: squint source mapping
 export default function viteSquint(opts = {}) {
   const squint = {
     name: "squint_compile",
-    transform: function (code, id) {
+    enforce: "pre",
+    async load(id) {
+      // the resolveId adds the .jsx extension
+      if (/\.cljs.jsx$/.test(id)) {
+        // for cljs files, we just need to load and compile
+        // TODO: macros
+        // TODO: squint source mapping
+        const file = id.replace(/.jsx$/, "");
+        const code = await fs.promises.readFile(file, "utf-8");
+        const compiled = compileString(code);
+        return { code: compiled, map: null };
+      }
+    },
+    resolveId(id, imported, options) {
       if (/\.cljs$/.test(id)) {
-        const cs = compileString(code);
-        const jsx = esbuild.transformSync(cs, {
-          loader: "jsx",
-          jsx: "automatic",
-        });
-        return {
-          code: jsx.code,
-          map: null,
-        };
+        // For cljs files we need to do the following:
+        // absolutize the path, this makes it easier for load and other plugins
+        // append .jsx so that other plugins can pick it up
+        const absolutePath = path.resolve(dirname(imported), id);
+        if (options.scan) {
+          // this is a scan, return virtual module
```

Comment:
Interesting, maybe add a link to vite docs here to explain?

Reply:
Updated the comment on this to explain why we do it